homeassistant/auth/mfa_modules/insecure_example.py

```diff
@@ -0,0 +1,84 @@
+"""Example auth module."""
+import logging
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from . import MultiFactorAuthModule, MULTI_FACTOR_AUTH_MODULES, \
+    MULTI_FACTOR_AUTH_MODULE_SCHEMA
+
+CONFIG_SCHEMA = MULTI_FACTOR_AUTH_MODULE_SCHEMA.extend({
+    vol.Required('users'): [vol.Schema({
+        vol.Required('user_id'): str,
+        vol.Required('pin'): str,
+    })]
+}, extra=vol.PREVENT_EXTRA)
+
+STORAGE_VERSION = 1
+STORAGE_KEY = 'mfa_modules.insecure_example'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@MULTI_FACTOR_AUTH_MODULES.register('insecure_example')
+class InsecureExampleModule(MultiFactorAuthModule):
+    """Example auth module validate pin."""
+
+    DEFAULT_TITLE = 'Insecure Personal Identify Number'
+
+    def __init__(self, hass, config):
+        """Initialize the user data store."""
+        super().__init__(hass, config)
+        self._data = None
+        self._users = config['users']
+
+    @property
+    def input_schema(self):
+        """Validate login flow input data."""
+        schema = OrderedDict()
+        schema['pin'] = str
+        return vol.Schema(schema)
+
+    @property
+    def setup_schema(self):
+        """Validate async_setup_user input data."""
+        schema = OrderedDict()
+        schema['pin'] = str
+        return vol.Schema(schema)
```

Comment:
So here it is getting kinda weird.

The way mfa modules work, is that we persist in the user which mfa are enabled, and then we ask the mfa module to show a form. I feel like this violates the single source of truth. We should probably not store which mfa modules are enabled for a user. Instead:

- MFA modules will get a method `async_is_enabled_for(user)`
- We can add a method to auth manager to gather enabled MFA modules for a user. It then asks each MFA module.

I think that since this module does not support setting up, it should return `None` for setup schema, which indicates to the frontend that this module cannot be setup.

What do you think of that?